packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -1358,6 +1358,152 @@ test.describe('goto', () => {
 			: 'goto: invalid URL';
 		await expect(page.locator('p')).toHaveText(message);
 	});
+
+	test.describe('navigation and redirects should be consistent between web native and sveltekit based', () => {
+		test.beforeEach(async ({ page }) => {
+			await page.goto('/goto/testentry');
+			await page.goto('/goto/teststart');
+
+			await expect(page).toHaveURL('/goto/teststart');
+		});
+
+		test.describe('navigating outside the app on sameorigin', () => {
+			[
+				[
+					'app.goto { replaceState: false }',
+					async ({ app }) => {
+						// navigating to nonexistent page causes playwright's page context to be destroyed
+						// thus this call throws an error unless caught
+						await app.goto('/goto/nonexistent', { replaceState: false }).catch(() => {});
+					}
+				],
+				[
+					'location.assign',
+					async ({ page }) => {
+						await page.evaluate(() => {
+							location.assign('/goto/nonexistent');
+						});
+					}
+				]
+			].forEach(([title, navigate]) => {
+				test(title, async ({ app, page }) => {
+					await navigate({ app, page });
+					await expect(page).toHaveURL('/goto/nonexistent');
+
+					await page.goBack();
+					await expect(page).toHaveURL('/goto/teststart');
+				});
+			});
```

Comment:
The tests are a bit difficult to read in this structure and it's also resulting in type errors due to TypeScript inferring the type of the array too loosely. Could we simplify them to adopt a structure similar to the other tests where we just duplicate some of the statements?